src/item.cpp

```diff
@@ -1648,30 +1648,32 @@ std::string item::info( bool showtext, std::vector<iteminfo> &info ) const
                                       _( "* This object is <neutral>surrounded</neutral> by a <info>sickly green glow</info>." ) ) );
         }
 
-        if( is_food() && has_flag( "CANNIBALISM" ) ) {
-            if( !g->u.has_trait_flag( "CANNIBAL" ) ) {
-                info.push_back( iteminfo( "DESCRIPTION",
-                                          _( "* This food contains <bad>human flesh</bad>." ) ) );
-            } else {
-                info.push_back( iteminfo( "DESCRIPTION",
-                                          _( "* This food contains <good>human flesh</good>." ) ) );
+        if( is_food() ) {
+            if( has_flag( "CANNIBALISM" ) ) {
+                if( !g->u.has_trait_flag( "CANNIBAL" ) ) {
+                    info.emplace_back( "DESCRIPTION", _( "* This food contains <bad>human flesh</bad>." ) );
+                } else {
+                    info.emplace_back( "DESCRIPTION", _( "* This food contains <good>human flesh</good>." ) );
+                }
             }
-        }
 
-        ///\EFFECT_SURVIVAL >=3 allows detection of poisonous food
-        if( is_food() && has_flag( "HIDDEN_POISON" ) && g->u.skillLevel( skill_survival ).level() >= 3 ) {
-            info.push_back( iteminfo( "DESCRIPTION",
-                                      _( "* On closer inspection, this appears to be <bad>poisonous</bad>." ) ) );
-        }
+            if( is_tainted() ) {
+                info.emplace_back( "DESCRIPTION", _( "* This food is <bad>tainted</bad> and will poison you." ) );
+            }
 
-        ///\EFFECT_SURVIVAL >=5 allows detection of hallucinogenic food
-        if( is_food() && has_flag( "HIDDEN_HALLU" ) && g->u.skillLevel( skill_survival ).level() >= 5 ) {
-            info.push_back( iteminfo( "DESCRIPTION",
-                                      _( "* On closer inspection, this appears to be <neutral>hallucinogenic</neutral>." ) ) );
+            ///\EFFECT_SURVIVAL >=3 allows detection of poisonous food
+            if( has_flag( "HIDDEN_POISON" ) && g->u.skillLevel( skill_survival ).level() >= 3 ) {
+                info.emplace_back( "DESCRIPTION", _( "* On closer inspection, this appears to be <bad>poisonous</bad>." ) );
+            }
+
+            ///\EFFECT_SURVIVAL >=5 allows detection of hallucinogenic food
+            if( has_flag( "HIDDEN_HALLU" ) && g->u.skillLevel( skill_survival ).level() >= 5 ) {
+                info.emplace_back( "DESCRIPTION", _( "* On closer inspection, this appears to be <neutral>hallucinogenic</neutral>." ) );
+            }
         }
 
-        if( ( is_food() && has_flag( "BREW" ) ) || ( is_food_container() &&
-                contents[0].has_flag( "BREW" ) ) ) {
+
+        if( ( is_food() && has_flag( "BREW" ) ) || ( is_food_container() && contents[0].has_flag( "BREW" ) ) ) {
```

Comment:
Now that I look at this section and the section above, they could be combined. The food part could be extracted (something like `const item &the_food = is_food() ? *this : contents[0]`), then the flags would show for contained food too.

Currently it's no big deal, but human bone broth already hides its cannibalism description when in a container.